src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1154,6 +1154,7 @@ public void addTab(BasePanel basePanel, boolean raisePanel) {
             // add tab
             Tab newTab = new Tab(basePanel.getTabTitle(), basePanel);
             tabbedPane.getTabs().add(newTab);
+            newTab.setOnCloseRequest(event -> closeTab((BasePanel) newTab.getContent()));
```

Comment:
That's a very good observation. 

As of now the user can close the tab even if there are unsaved changes. Thus, we should improve the functionality here a bit by calling `confirmClose` and consuming the `event` in case the user canceled the close operation.

Reply:
The method `closeTab()` makes a call to `confirmClose` when the panel is modified. However, it does not consume the `event` in case of cancelling. Since, the logic for removing tab is already handled I think it should be enough to consume the event in the `closeRequest` Itself.

Reply:
The `confirmClose` shows a dialog with 3 option *Save changes*, *Discard changes* and *Return to JabRef*. The `confirmClose` method returns false only if the user clicks *Return to JabRef* in this case, the tab isn't closed however the autosave and backup managers are shutdown any way. 
I don't think they should be shutdown in this case, we can return from `closeTab()` in case `confirmClose` returns false.